sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -10,3 +10,175 @@
 
 #include <_az_cfg.h>
 
+static const az_span provisioning_service_api_version
+    = AZ_SPAN_LITERAL_FROM_STR("/api-version=" AZ_IOT_PROVISIONING_SERVICE_VERSION);
+
+static const az_span str_dps = AZ_SPAN_LITERAL_FROM_STR("$dps");
+static const az_span str_registrations = AZ_SPAN_LITERAL_FROM_STR("/registrations/");
+static const az_span str_put_iotdps_register
+    = AZ_SPAN_LITERAL_FROM_STR("PUT/iotdps-register/?$rid=1");
+static const az_span str_get_iotdps_get_operationstatus
+    = AZ_SPAN_LITERAL_FROM_STR("GET/iotdps-get-operationstatus/?$rid=1&operationId=");
+static const az_span str_res = AZ_SPAN_LITERAL_FROM_STR("res/");
+
+AZ_NODISCARD az_iot_provisioning_client_options az_iot_provisioning_client_options_default()
+{
+  return (az_iot_provisioning_client_options){ .user_agent = AZ_SPAN_NULL };
+}
+
+AZ_NODISCARD az_result az_iot_provisioning_client_init(
+    az_iot_provisioning_client* client,
+    az_span global_device_endpoint,
+    az_span id_scope,
+    az_span registration_id,
+    az_iot_provisioning_client_options const* options)
+{
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION_VALID_SPAN(global_device_endpoint, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(id_scope, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(registration_id, 1, false);
+
+  client->_internal.global_device_endpoint = global_device_endpoint;
+  client->_internal.id_scope = id_scope;
+  client->_internal.registration_id = registration_id;
+
+  if (options != NULL)
+  {
+    client->_internal.options.user_agent = options->user_agent;
```

Comment:
nit: should you just assign the options to the options? This prevents needing to update it every time an option is added.
```suggestion
    client->_internal.options = options;
```

Reply:
or i might have ran into this myself. does the const disallow this?

Reply:
Copy paste from the telemetry code :D.
I'll give it a try - if it works, I'll change both.